src/gurt/misc.c

```diff
@@ -566,3 +567,87 @@ void d_getenv_int(const char *env, unsigned *int_val)
 	D_DEBUG(DB_TRACE, "d_getenv_int(), get ENV %s as %d.\n", env, value);
 	*int_val = value;
 }
+
+/**
+ * Write formatted data to d_string_buffer_t
+ *
+ * Create a string with the same text that would be created by printf(). The
+ * result string is stored as a C string in a buffer pointed by buf. The
+ * d_string_buffer_t internal buffer grows as text is written.
+ *
+ * \param[in,out] buf    string object where the formatted string will be
+ *                       stored. Subsequent write operations to the same string
+ *                       object will be append at the end of the buffer.
+ * \param[in]     format this is the string that contains the text to be
+ *                       written to d_string_buffer_t. Please refer to the
+ *                       printf() documentation for full details of how to use
+ *                       the format tags.
+ * \return               0 on success, errno code on failure.
```

Comment:
It also returns positive values in places.

Reply:
My bad, it should return zero rather than positive number on success. Or even better as Alexander suggested `DER_SUCCESS`